vllm/v1/core/scheduler.py

```diff
@@ -507,23 +515,27 @@ def update_from_output(
             # Transmit partial if chunked prefill & prompt logprobs is enabled
             if new_token_ids or prompt_logprobs_tensors is not None:
                 # Add EngineCoreOutput for this Request.
-                outputs.append(
-                    EngineCoreOutput(
-                        request_id=req_id,
-                        new_token_ids=new_token_ids or [],
-                        finish_reason=request.get_finished_reason(),
-                        new_logprobs=new_logprobs,
-                        new_prompt_logprobs_tensors=prompt_logprobs_tensors,
-                        stop_reason=request.stop_reason))
+                # Add EngineCoreOutputs for this Request.
+                output.request_ids.append(req_id)
+                output.new_token_id_offsets.append(offset)
+                new_ids = request.output_token_ids[-num_new_tokens:]
+                output.new_token_ids += new_ids
```

Comment:
It would be for another iteration, but I'm thinking he we may want to do this outside of the loop, keep the new token ids as a tensor and either send as-is and do additional filtering in the front-end process, or do the filtering via tensor slicing/index_select type operations.

This would have the benefits of:
- Eliminating intermediate objects creation which scales with the batch size
- Eliminating serialization overhead - I think if we can keep as much as possible of EngineCoreOutputs in tensor/numpy form, we can transmit the backing buffer(s) as-is ... zmq can read directly from these and later we could also see if we can use shm
- Moving some work to the front-end process which we can more easily scale out